Samples/MapScenarios/MapScenarios/Scenarios/VoicedRouteViewController.swift

```diff
@@ -0,0 +1,169 @@
+        DispatchQueue.main.async { [weak self] in
+            if let buildingIdentifier = self?.buildingIdentifier {
+                let managedLocationManager = PWManagedLocationManager(buildingId: buildingIdentifier)
+                self?.mapView.register(managedLocationManager)
+            }
+        }
+    }
+    
+    // MARK: Voice Prompts Button
+    
+    func configureSpeechButton() {
+        navigationItem.rightBarButtonItem = UIBarButtonItem(title: "Speech", style: .plain, target: self, action: #selector(speechButtonTapped))
+        updateSpeechButton()
+    }
+    
+    func updateSpeechButton() {
+        navigationItem.rightBarButtonItem?.tintColor = speechEnabled ? .blue : .darkGray
+    }
+    
+    @objc
+    func speechButtonTapped() {
+        speechEnabled = !speechEnabled
+        updateSpeechButton()
+    }
+}
+
+// MARK: - PWMapViewDelegate
+
+extension VoicedRouteViewController: PWMapViewDelegate {
+    
+    func mapView(_ mapView: PWMapView!, locationManager: PWLocationManager!, didUpdateIndoorUserLocation userLocation: PWUserLocation!) {
+        if !firstLocationAcquired {
+            firstLocationAcquired = true
+            mapView.trackingMode = .follow
+            
+            let destinationPOIIdentifier = 0 /* Replace with the destination POI identifier */
+            
+            var destinationPOI: PWPointOfInterest!
+            if destinationPOIIdentifier != 0 {
+                destinationPOI = mapView.building.pois.first(where: { $0.identifier == destinationPOIIdentifier })
+            } else {
+                if let firstPOI = mapView.building.pois.first {
+                    destinationPOI = mapView.building.pois[2]
+                }
+            }
+            
+            if destinationPOI == nil {
+                print("No points of interest found, please add at least one to the building in the Maas portal")
+                return
+            }
+            
+            PWRoute.createRoute(from: mapView.indoorUserLocation, to: destinationPOI, accessibility: false, excludedPoints: nil, completion: { (route, error) in
+                guard let route = route else {
+                    print("Couldn't find a route from you current location to the destination.")
+                    return
+                }
+                
+                mapView.navigate(with: route)
+            })
+        }
+    }
+    
+    func mapView(_ mapView: PWMapView!, didChange instruction: PWRouteInstruction!) {
+        guard let movement = instruction.movement, speechEnabled, !previouslyReadInstructions.contains(instruction) else {
+            return
+        }
+        previouslyReadInstructions.insert(instruction)
+        var voicePrompt = "\(movement)"
+        if let turn = instruction.turn {
+            voicePrompt = voicePrompt + ", then \(turn)"
+        }
+        let utterance = AVSpeechUtterance(string: voicePrompt)
+        utterance.voice = AVSpeechSynthesisVoice(language: "en-us")
```

Comment:
Should this be hard-coded to english or can we show the example of using the device's current language? Would be a more globalized solution.